### PR TITLE
Covariance calculation functions added to RiskEstimators module

### PR DESCRIPTION
**Pull request by PanPip**
# Description

This is the addition of functions related to the calculation of Covariance Matrices and their adjustments to the RiskEstimators class. 

The following functions are added:
- Minimum Covariance Determinant
- Maximum likelihood covariance estimator (Empirical covariance)
- Covariance estimator with shrinkage
- Semi-Covariance matrix
- Exponentially-weighted Covariance matrix

The unit test and Sphinx documentation are also provided with the code. The notebook with examples of use is in the H&T research repository.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] test_risk_estimators.py

**Test Configuration**:
* Windows 10
* PyCharm

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mlfinlab/portfolio_optimization/risk_estimators.py [hunk 1]
```diff
@@ -258,3 +260,197 @@ def denoise_covariance(self, cov, tn_relation, kde_bwidth):
...
+
+    @staticmethod
+    def minimum_covariance_determinant(returns, price_data=False, assume_centered=False,
+                                       support_fraction=None, random_state=None):
+        """
+        Calculates the Minimum Covariance Determinant for a dataframe of asset prices or returns.
+
+        This function is a wrap of the sklearn's MinCovDet (MCD) class.
+
+        :param returns: (pd.dataframe) Dataframe where each column is a series of returns or prices for an asset.
```
Comment (Jackal08) [inline mlfinlab/portfolio_optimization/risk_estimators.py:272]: Under what conditions would you build a cov mat on prices? Are these prices then transformed to returns or it uses raw prices?

Comment (Jackal08) [reply mlfinlab/portfolio_optimization/risk_estimators.py:272]: It looks like it transforms price to rets. You should specify in the doc string that it applies that transformation.

File: mlfinlab/portfolio_optimization/risk_estimators.py [hunk 2]
```diff
@@ -258,3 +260,197 @@ def denoise_covariance(self, cov, tn_relation, kde_bwidth):
         cov_denoised = self.corr_to_cov(corr, np.diag(cov) ** (1 / 2))
 
         return cov_denoised
+
+    @staticmethod
+    def minimum_covariance_determinant(returns, price_data=False, assume_centered=False,
+                                       support_fraction=None, random_state=None):
+        """
+        Calculates the Minimum Covariance Determinant for a dataframe of asset prices or returns.
```
Comment (Jackal08) [inline mlfinlab/portfolio_optimization/risk_estimators.py:268]: Add more detailed docstings that also explain the method and link to supporting documentation such as sklearn. Its ok to say that this function is a wrapper for sklean.

File: mlfinlab/portfolio_optimization/risk_estimators.py (hunk 1 above)
Comment (PanPip) [reply mlfinlab/portfolio_optimization/risk_estimators.py:272]: Will do

File: mlfinlab/portfolio_optimization/risk_estimators.py (hunk 2 above)
Comment (PanPip) [reply mlfinlab/portfolio_optimization/risk_estimators.py:268]: Will do

**Pull request merged**